sync/sync.go

```diff
@@ -238,7 +238,9 @@ func (s *Syncer[H]) doSync(ctx context.Context, fromHead, toHead H) (err error)
 
 	s.stateLk.Lock()
 	s.state.End = time.Now()
-	s.state.Error = err
+	if err != nil {
+		s.state.Error = err.Error()
+	}
```

Comment:
```suggestion
	if err != nil {
		s.state.Error = err.Error()
	} else {
		s.state.Error = “”
   }
```

Otherwise, we don't override the error. 
 
Found by @oblique